patterns/multiparty-note-sharing/broadcaster/src/test/e2e.test.ts

```diff
@@ -0,0 +1,118 @@
+import {
+    AccountWalletWithPrivateKey,
+    AztecAddress,
+    computeAuthWitMessageHash,
+    createPXEClient,
+    ExtendedNote,
+    Fr,
+    PXE,
+    waitForPXE,
+  } from "@aztec/aztec.js";
+  
+  import { BroadcasterContract } from "../../../../../artifacts/multiparty-note-sharing/broadcaster/Broadcaster.js";
+  import { createAccount, getInitialTestAccountsWallets } from "@aztec/accounts/testing";
+  
+  // Global variables
+  let pxe: PXE;
+  let broadcaster: BroadcasterContract;
+  let sharedKey: Fr;
+  
+  let user: AccountWalletWithPrivateKey;
+  let user1: AccountWalletWithPrivateKey;
+  let user2: AccountWalletWithPrivateKey;
+  let user3: AccountWalletWithPrivateKey;
+  let user4: AccountWalletWithPrivateKey;
+  let deployer: AccountWalletWithPrivateKey;
+  let wallets: AccountWalletWithPrivateKey[];
+  let accounts: AztecAddress[];
+  
+  const { PXE_URL = 'http://localhost:8080' } = process.env;
+  
+  const setupSandbox = async () => {
+    const pxe = createPXEClient(PXE_URL);
+    await waitForPXE(pxe);
+    return pxe;
+  };
+  
+  // Setup: Set the sandbox
+  beforeAll(async () => {
+    pxe = await setupSandbox();
+    [user, user1, deployer] = await getInitialTestAccountsWallets(pxe);
+    user2 = await createAccount(pxe);
+    user3 = await createAccount(pxe);
+    user4 = await createAccount(pxe);
```

Comment:
maybe worth noting somewhere that we have the pk in the pxe due to this createAccount(...)